src/transformers/models/blip_2/modeling_blip_2.py

```diff
@@ -1549,6 +1830,432 @@ def forward(
+
+        >>> model = Blip2ModelWithoutLM.from_pretrained("jpizarrom/blip2-itm-vit-g", torch_dtype=torch.float16)
+        >>> processor = AutoProcessor.from_pretrained("jpizarrom/blip2-itm-vit-g")
+
+        >>> model.to(device)  # doctest: +IGNORE_RESULT
+
+        >>> url = "http://images.cocodataset.org/val2017/000000039769.jpg"
+        >>> image = Image.open(requests.get(url, stream=True).raw)
+
+        >>> inputs = processor(
+        ...     text=["a photo of a cat", "a photo of a dog"], images=image, return_tensors="pt", padding=True
+        ... ).to(device, torch.float16)
+
+        >>> outputs = model(**inputs)
+        >>> logits_per_image = outputs.logits_per_image  # this is the image-text similarity score
+        >>> probs = logits_per_image.softmax(dim=1)  # we can take the softmax to get the label probabilities
+        ```"""
+        # Use BLIP2 model's config for some fields (if specified) instead of those of vision & text components.
+        output_attentions = output_attentions if output_attentions is not None else self.config.output_attentions
+        output_hidden_states = (
+            output_hidden_states if output_hidden_states is not None else self.config.output_hidden_states
+        )
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+
+        vision_outputs = self.vision_model(
+            pixel_values=pixel_values,
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            return_dict=return_dict,
+        )
+
+        image_embeds = vision_outputs[0]
+        image_attention_mask = torch.ones(image_embeds.size()[:-1], dtype=torch.long, device=image_embeds.device)
+
+        query_tokens = self.query_tokens.expand(image_embeds.shape[0], -1, -1)
+        query_outputs = self.qformer(
+            query_embeds=query_tokens,
+            encoder_hidden_states=image_embeds,
+            encoder_attention_mask=image_attention_mask,
+            return_dict=return_dict,
+        )
+
+        image_embeds = query_outputs[0] if not return_dict else query_outputs.last_hidden_state
+        image_embeds = self.vision_proj(image_embeds)
+
+        question_outputs = self.qformer(
+            input_ids=input_ids,
+            attention_mask=attention_mask,
+            return_dict=return_dict,
+        )
+        question_embeds = question_outputs[0] if not return_dict else question_outputs.last_hidden_state
+        question_embeds = self.text_proj(question_embeds[:, 0, :])
+
+        image_feat = normalize(image_embeds, dim=-1)
+        text_feat = normalize(question_embeds, dim=-1)
+
+        # text-query similarity
+        sim_t2q = torch.matmul(text_feat.unsqueeze(1).unsqueeze(1), image_feat.permute(0, 2, 1)).squeeze(2)
+
+        # text-image similarity: aggregate across all query tokens
+        logits_per_text, _ = sim_t2q.max(-1)
+        logits_per_text = logits_per_text / self.temp
```

Comment:
that code was removed, `Blip2ModelWithProjection` class was completely removed, maybe it could be added in other PR (`Blip2ForPreTraining` could be a better class name)